config/core/300-resources/service.yaml

```diff
@@ -59,6 +59,1129 @@ spec:
   - << : *version
     name: v1
     storage: true
+    schema:
+      openAPIV3Schema:
+        x-kubernetes-preserve-unknown-fields: true
```

Comment:
does this apply to nested properties now?

Reply:
I think it applies to everything under it. I left it in place because I don't have a way to verify that I have a perfect match to the API.

Reply:
https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#controlling-pruning

Reply:
Digging into the example you linked it seems like `spec.something` was pruned.

> This means that the something field in the specified spec object is pruned, but everything outside is not.


It seems like `x-kubernetes-preserve-unknown-fields` doesn't apply to nested properties

Reply:
I'll admit that I'm not entirely sure what it does. I thought it effected the whole tree based on 
`By default, all unspecified fields for a custom resource, across all versions, are pruned. It is possible though to opt-out of that for specifc sub-trees of fields by adding x-kubernetes-preserve-unknown-fields: true`

Reply:
> for specifc sub-trees of fields

it seems like if you have a nested `properties` attribute pruning is enabled again

Reply:
thus I wonder if we _might_ need `x-kubernetes-preserve-unknown-fields` in a few places if we aren't specifying the full schema

Reply:
yeah if the full schema is present we _shouldn't_ need it